### ISSUE TITLE
Error should be an actual `Error`

### ISSUE THREAD

**Issue by sindresorhus** [sass/node-sass#675]
Currently node-sass returns an object in the `error` callback. This is awkward and it should rather be an instance of `Error`.

Would also be nice if `err.toString()` returned a formatted error message so I don't have to do that manually.

Something like this (but also with filename):

![image](https://cloud.githubusercontent.com/assets/170270/6185814/502fc4c2-b3a5-11e4-9072-41571c128d20.png)

*(from [`gulp-jscs`](https://github.com/jscs-dev/gulp-jscs))*

**Comment by am11** [comment]
This change is by design. Our error is a JSON object containing message (string) and error code (integer).

**Comment by sindresorhus** [comment]
I know it's by design, but IMHO the wrong one.

**Comment by kevva** [comment]
Yeah, should just remove `JSON.parse()` here https://github.com/sass/node-sass/blob/master/lib/index.js#L150.

**Comment by am11** [comment]
@kevva can you give me a reason why would anyone dislike `error.message`? 

We removed result parsing in success cb from v2.0.0 (which was introduced in 2.0.0-beta), because that was an unnecessary overhead (related: https://github.com/sass/node-sass/issues/623#issuecomment-69873321). 

For errors, however, we are not only parsing it to inject `code` in the object literal, but we are encouraging the callback to accept the useful object (instead of a JSON string -- which they will parse in majority of cases to read the message -- with second integer `code` argument). If the downstream implementer want it otherwise, they have cast it the way they want. IMVHO this default makes perfect sense (unless of course there are good reason).

**Comment by kevva** [comment]
@am11, the error object uses `err.message` by default.

**Comment by am11** [comment]
:+1: Makes sense! And it will be a non-breaking change i.e.:

```javascript
function mySassErrorHandled(err) {  // callback for node-sass
    writeToFile(err.code + ': ' + err.message);
}
```

**Comment by am11** [comment]
This was addressed by #668.
